lib/ads/interstitial_ad_manager.js

```diff
@@ -280,7 +285,9 @@ shaka.ads.InterstitialAdManager = class {
       }).length;
     }
 
-    this.interstitials_.delete(interstitial);
+    if (interstitial.once) {
+      this.interstitials_.delete(interstitial);
```

Comment:
Perhaps I am misunderstanding how this would work with `numberToSkip`, but I feel like this would cause problems if there are multiple `ONCE` interstitials in a row?

For example, imagine there are two `ONCE` interstitials and nothing else.
We will start out with the interstitial at `adPosition` 0 first, and then to find the next interstitial we will look for the interstitial at `adPosition` 1.
But if we delete the interstitial at 0, won't the list of interstitials that `getCurrentIntestitial_` looks at be only a single element long? So skipping the first interstitial will just cause it to find no interstitial at all, right, and skip over the second interstitial entirely?

Maybe I am not seeing something here.

Reply:
You are right...